http3/server.go

```diff
@@ -149,6 +149,9 @@ func (s *Server) serveImpl(tlsConf *tls.Config, conn net.PacketConn) error {
 					return nil, err
 				}
 			}
+			if conf == nil {
+				return nil, nil
+			}
```

Comment:
NextProtos may not be set if return immediately?

Reply:
Good catch. Should be fixed now.